modules/alia-component/src/qbits/alia/component.clj

```diff
@@ -0,0 +1,133 @@
+
+(defn ->opts
+  "This should provide a solid way to parameterize interaction with
+   Cassandra. For each named query `nick` we determine what options to
+   give to the query in the following way:
+
+   - We fetch defaults from the configuration provided query options
+   - We merge code-local defaults for this query
+   - We merge query-specific defaults from configuration"
+  [nick config code-opts]
+  (let [opts (merge (:defaults config) code-opts (get config nick))]
+    (s/assert ::alia/execute-opts
+              (reduce-kv #(cond-> %1 %3 (assoc %2 %3)) {} opts))))
+
+(defn ->query
+  "Get a function which executes a query, preparing the statement."
+  [nick session query-opts {:keys [statement opts]}]
+  (let [prepared (alia/prepare session statement)
+        cfg-opts (->opts nick query-opts opts)]
+    (fn [run-opts]
+      (let [opts (merge cfg-opts run-opts)
+            exec-fn (or (:execute-fn opts) alia/execute)]
+        (exec-fn session prepared (dissoc opts :execute-fn))))))
+
+(defn query-fn-map
+  "Yield a map of query nickname to function executing the query.
+   Provided queries should be statements valid for `alia/prepare`."
+  [session queries query-opts]
+  (reduce-kv #(assoc %1 %2 (->query %2 session query-opts %3)) {} queries))
+
+(defn cluster-query-opts
+  "Fetch query-options from configuration."
+  [registry cluster-name]
+  (let [globals (:query-opts registry)
+        cluster (get-in registry [:cluster-opts cluster-name])]
+    (merge globals cluster)))
+
+(defn query-functions
+  "Given a cluster, keyspace, and query map, yield a function
+   which executes queries by nickname. If no cluster-name
+   is provided, assume a single-cluster registry and fetch
+   the first from the registry.
+
+   The query function has three arities: 1, 2, and 3.
+   The first argument is always the nickname, the optional
+   second one is a vector of values to attach to the prepared
+   query, the last one additional options to override when
+   sending the query.
+
+   Here, options accept an addition `:execute-fn` key which
+   determine which function should be used to send the
+   query out, defaulting to `alia/execute`.
+
+   Each query can be configured with global configuration defaults,
+   code-supplied defaults, and query-specific configuration overrides,
+   processed in that order as documented in `query`."
+  ([registry cluster-name keyspace queries]
+   (let [session (keyspace-session registry cluster-name keyspace)
+         opts    (cluster-query-opts registry cluster-name)
+         qmap    (query-fn-map session queries opts)]
+     (swap! (:sessions registry) conj session)
+     (fn execute-query
+       ([qname]
+        (execute-query qname [] {}))
+       ([qname values]
+        (execute-query qname values {}))
+       ([qname values run-opts]
+        (if-let [query-fn (get qmap qname)]
+          (query-fn (cond-> run-opts (seq values) (assoc :values (vec values))))
```

Comment:
I wouldn't coerce :values to vec, since it could be a hash-map for named placeholders : 

ex 
```clj
(execute s "select * from foo where id= :something" {:values {:something "bar"}})
```